storage: TestNodeLivenessSelf failed under stress

SHA: https://github.com/cockroachdb/cockroach/commits/4a9b6b5fd6969b9ac72671af590f0f06cc97cbe2

Parameters:
```
TAGS=
GOFLAGS=-race
```

Stress build found a failed test: https://teamcity.cockroachdb.com/viewLog.html?buildId=388769&tab=buildLog

```

I171025 08:25:10.585901 105201 util/stop/stopper.go:476  quiescing; tasks left:
2      [async] storage.multiTestContextKVTransport: calling next replica
I171025 08:25:10.589242 105201 util/stop/stopper.go:476  quiescing; tasks left:
1      [async] storage.multiTestContextKVTransport: calling next replica
W171025 08:25:10.597474 105289 vendor/google.golang.org/grpc/transport/log.go:36  transport: http2Server.HandleStreams failed to read frame: read tcp 127.0.0.1:40187->127.0.0.1:36567: use of closed network connection
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x0 pc=0x210608a]

goroutine 105069 [running]:
testing.tRunner.func1(0xc42061a870)
	/usr/local/go/src/testing/testing.go:711 +0x5d9
panic(0x24ffb80, 0x38d9530)
	/usr/local/go/src/runtime/panic.go:491 +0x2a2
github.com/cockroachdb/cockroach/pkg/util/leaktest.AfterTest.func1()
	/go/src/github.com/cockroachdb/cockroach/pkg/util/leaktest/leaktest.go:69 +0x551
panic(0x24ffb80, 0x38d9530)
	/usr/local/go/src/runtime/panic.go:491 +0x2a2
github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessSelf.func2(0x22d9bdaf, 0xed1824166)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:406 +0xda
github.com/cockroachdb/cockroach/pkg/util.RetryForDuration(0xa7a358200, 0xc42138df30, 0xc420c56501, 0xc42036d8c0)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/retry.go:31 +0xda
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoonDepth(0x1, 0x392dfe0, 0xc42061a870, 0xc42138df30)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:42 +0x60
github.com/cockroachdb/cockroach/pkg/testutils.SucceedsSoon(0x392dfe0, 0xc42061a870, 0xc42138df30)
	/go/src/github.com/cockroachdb/cockroach/pkg/testutils/soon.go:36 +0x56
github.com/cockroachdb/cockroach/pkg/storage_test.TestNodeLivenessSelf(0xc42061a870)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/node_liveness_test.go:405 +0x613
testing.tRunner(0xc42061a870, 0x27d8f50)
	/usr/local/go/src/testing/testing.go:746 +0x16d
created by testing.(*T).Run
	/usr/local/go/src/testing/testing.go:789 +0x569

goroutine 1 [chan receive]:
testing.(*T).Run(0xc4201dc3c0, 0x270c4c6, 0x14, 0x27d8f50, 0xc420659901)
	/usr/local/go/src/testing/testing.go:790 +0x59b
testing.runTests.func1(0xc4201dc3c0)
	/usr/local/go/src/testing/testing.go:1004 +0xa8
testing.tRunner(0xc4201dc3c0, 0xc420659b00)
	/usr/local/go/src/testing/testing.go:746 +0x16d
testing.runTests(0xc4202b3f00, 0x38f9580, 0x1d8, 0x1d8, 0x5d)
	/usr/local/go/src/testing/testing.go:1002 +0x522
testing.(*M).Run(0xc4208ddf20, 0x0)
	/usr/local/go/src/testing/testing.go:921 +0x207
github.com/cockroachdb/cockroach/pkg/storage_test.TestMain(0xc420659f20)
	/go/src/github.com/cockroachdb/cockroach/pkg/storage/main_test.go:57 +0x2c7
main.main()
	github.com/cockroachdb/cockroach/pkg/storage/_test/_testmain.go:1002 +0x1ce

goroutine 3 [chan receive]:
github.com/cockroachdb/cockroach/pkg/util/log.(*loggingT).flushDaemon(0x3e13220)
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:1005 +0x8f
created by github.com/cockroachdb/cockroach/pkg/util/log.init.0
	/go/src/github.com/cockroachdb/cockroach/pkg/util/log/clog.go:581 +0x13a

goroutine 4 [syscall, 4 minutes]:
os/signal.signal_recv(0x727501)
	/usr/local/go/src/runtime/sigqueue.go:131 +0xa6
os/signal.loop()
	/usr/local/go/src/os/signal/signal_unix.go:22 +0x30
created by os/signal.init.0
	/usr/local/go/src/os/signal/signal_unix.go:28 +0x4f

goroutine 30545 [select, locked to thread]:
runtime.gopark(0x27db120, 0x0, 0x26fb269, 0x6, 0x18, 0x1)
	/usr/local/go/src/runtime/proc.go:277 +0xfd
runtime.selectgo(0xc420996750, 0xc420b19bc0)
	/usr/local/go/src/runtime/select.go:395 +0x12c3
runtime.ensureSigM.func1()
	/usr/local/go/src/runtime/signal_unix.go:511 +0x1a7
runtime.goexit()
	/usr/local/go/src/runtime/asm_amd64.s:2337 +0x1


ERROR: exit status 2

context canceled
make: *** [stress] Error 1
9 runs completed, 1 failures, over 11m6s
Makefile:272: recipe for target 'stress' failed
```

--- a/pkg/storage/node_liveness_test.go
+++ b/pkg/storage/node_liveness_test.go
@@ -390,7 +390,10 @@ func TestNodeLivenessSelf(t *testing.T) {
 	// Verify liveness of all nodes for all nodes.
 	pauseNodeLivenessHeartbeats(mtc, true)
 	g := mtc.gossips[0]
-	liveness, _ := mtc.nodeLivenesses[0].GetLiveness(g.NodeID.Get())
+	liveness, err := mtc.nodeLivenesses[0].GetLiveness(g.NodeID.Get())
+	if err != nil {
+		t.Fatal(err)
+	}
 	if err := mtc.nodeLivenesses[0].Heartbeat(context.Background(), liveness); err != nil {
 		t.Fatal(err)
 	}

